coredns/plugin/parse.go

```diff
@@ -151,9 +153,5 @@ func parseSegments(segs []string, count int, r *recordRequest, qType uint16) (*r
 
 // stripUnderscore removes a prefixed underscore from s.
 func stripUnderscore(s string) string {
-	if s[0] != '_' {
-		return s
-	}
-
-	return s[1:]
+	return strings.TrimLeft(s, "_")
```

Comment:
```suggestion
	return strings.TrimPrefix(s, "_")
```

to match the existing behaviour (except with empty strings).